### DISCUSSION TITLE
Negative values

### DISCUSSION THREAD

**Discussion by ovahead** [mytourbook/mytourbook#1149] [General]
On some trips I have negative values, mainly on short trips with long breaks. However, these are marked as such in the device by interrupting the journey and continuing afterwards (Garmin Edge 830).
So what can I do to avoid this, so that MTB recognizes it correctly? Changing the pauses setting in MTB doesn't help.

Example 1 (10.07.23)
![Minuswerte](https://github.com/mytourbook/mytourbook/assets/112829463/281e1adc-a3fa-4ace-9fe8-a0434ac9b3e4)
In Garmin Connect:
Zeit
8:21.1
Zeit in Bewegung
8:11
Verstichene Zeit
55:01
Durchschnittsgeschwindigkeit 
17,5 km/h
Ø Geschwindigkeit in Bewegung
17,9
Maximale Geschwindigkeit
25,4

Example 2 (07.07.23)
![Minuswerte2](https://github.com/mytourbook/mytourbook/assets/112829463/1efeb4fb-c89e-4a9e-a629-ce74b0f1b2bf)
In Garmin Connect:
Zeit 16:23
Zeit in Bewegung
15:47
Verstrichene Zeit
1:29:20
Durchschnittsgeschwindigkeit
17,3
Ø Geschwindigkeit in Bewegung
17,9
Maximale Geschwindigkeit
31,7

**Comment by FJBDev** [comment]
Can you share 1or more files for which you see this issue ?

**Comment by ovahead** [comment]
Is there any way I can send you the files from the above issues privately?
I dont want to share them for the whole community ;)

**Reply by FJBDev** [reply]
Not sure how I can share my email without putting it publicly....

**Comment by ovahead** [comment]
Hm, What about a temporary  or throw away e-mail address? To get files with it should work.
https://muellmail.com/
https://www.trash-mail.com/

**Reply by FJBDev** [reply]
I thought about that too ([www.mailinator.com](https://www.mailinator.com)) but they're public so that doesn't solve the initial problem

**Comment by ovahead** [comment]
At trash-mail (english/german) you get your own mailbox for free, which you can also protect with a password. I haven't tested it myself, just read about it.
Account/Create new Mail-Adress and under the adress you can enter also a passwort. But must be activated. 
 
Otherwise create a free e-mail account at gmx.de/.net or web.de with a fake name, adress, possibly telefon number...

**Reply by FJBDev** [reply]
You can send me the file at this email

**Reply by ovahead** [reply]
You have an E-mail ;)

**Comment by FJBDev** [comment]
@ovahead 

Below are the results of my investigation.

>Since I edited them myself with FFRT before importing them into MyTourbook, I have included the originals. Don't think it makes a difference in MTB.

I am glad you mentioned that and that you included the original files because **yes**, it makes a difference in MT

The original files all have correct time values whereas the modified files all have negative pause values.

My hunch is that when you edited them in "FFRT", then it created the negative values.

I hope that helps!

Fixed files **with negative values**

![image](https://github.com/mytourbook/mytourbook/assets/25235665/d7b2b5c2-2219-4ba6-9ee8-968e992e084a)
![image](https://github.com/mytourbook/mytourbook/assets/25235665/17a69f1a-c048-4683-9d21-6482d3e147c2)

Original files **without negative values**
![image](https://github.com/mytourbook/mytourbook/assets/25235665/29bcd6f7-932a-455f-9172-83d11eb49f6c)
![image](https://github.com/mytourbook/mytourbook/assets/25235665/fc8655ce-078f-4301-9975-fd66b3900bc1)

**Reply by ovahead** [reply]
You're right! I think I have also some tours they also got negative values even I haven't edited them. I'll see if I can't find some ;)

I just only removed some very few entries with no speed values and added a start/stop event for them. At the beginning, at the point where the tour was interrupted anyway with the Garmin and maybe at the end. And the point where the tour was interrupted with the Garmin I then changed also the altitude. Just like I do on **every very small tour** with long breaks and the altitude has changed quite a lot, even though I haven't moved with the bike.
A few imported tours then have these negative values, but most are fine... 

Could it be that MTB is not interpreting some start/stop events correctly? Or of course Fit File Repair Tool doesn't generate them properly in a few cases...

**Reply by FJBDev** [reply]
> You're right! I think I have also some tours they also got negative values even I haven't edited them. I'll see if I can't find some ;)

That would be useful if you can find some

> Could it be that MTB is not interpreting some start/stop events correctly? Or of course Fit File Repair Tool doesn't generate them properly in a few cases...

I think it could be either or. I plan on doing a finer analysis soon and start first by checking that the edited files have the paused times created properly

**Reply by ovahead** [reply]
Would it be helpful if I sent you a similarly edited FIT file with the original where this minus value effect doesn't happen?

**Reply by FJBDev** [reply]
Here is the result of my analysis:
Looking and comparing at the original file vs the ones you modified, it seems to me that the pauses are badly formed in the "fixed" file and therefore create this negative value

Example with the file `2023-07-07-16-05-56-fixed.fit` (times are in GMT):
The 7th pause starts **Friday, July 7, 2023 2:11:36 PM** and ends **Friday, July 7, 2023 2:45:01 PM**
The 8th pause starts : **Friday, July 7, 2023 2:11:38 PM** and ends **Friday, July 7, 2023 2:45:00 PM**

Why is there a pause within a pause ? (btw, the 8th pause looks like it was added manually while the 7th one was in the original file)
While I can't answer this question as I don't know what type of modifications you have done, it doesn't make sense and seems like the file is not properly modified/edited.
This seems to be why MT computes negative paused time because of those duplicated/encapsulated pauses.

My recommendation would be to not duplicate and encapsulate pauses and it should fix your problem

**Reply by ovahead** [reply]
Okay, thank you very much for your work! I will do it.

I have a similar case here. If I only merged two tours (as round 1 and round 2) with FFRT because the tour was accidentally completely interrupted, then I also get minus values in MT.
I only merged them together with FIT File Repair Tool, otherwise I haven't changed anything...
So if you are also inrested to have a look at it, I'll be happy to send them to you ;)

**Comment by FJBDev** [comment]
> I have a similar case here. If I only merged two tours (as round 1 and round 2) with FFRT because the tour was accidentally completely interrupted, then I also get minus values in MT. I only merged them together with FIT File Repair Tool, otherwise I haven't changed anything... So if you are also inrested to have a look at it, I'll be happy to send them to you ;)

I can look at it if you want. But are you aware you can [merge tours i](https://mytourbook.sourceforge.io/mytourbook/index.php/documentation/tools/concatenate-tours)n MT directly ?

**Reply by ovahead** [reply]
Oops, I wasn't aware of that. Thank you very much for mentioning it! Then I should probably better use this function ;)
But there is no way to adjust the altitude of the second tour to the end of first tour, is there?

Anyway I sent you the files.

**Reply by FJBDev** [reply]
> Oops, I wasn't aware of that. Thank you very much for mentioning it! Then I should probably better use this function ;) But there is no way to adjust the altitude of the second tour to the end of first tour, is there?

Did you look at the [Adjust altitude documentation](https://mytourbook.sourceforge.io/mytourbook/index.php/documentation/tools/adjust-altitude) ?


> Anyway I sent you the files.

Thanks, I will look at them

**Reply by ovahead** [reply]
I wasn't sure what the effect of being able to correct the altitude evenly for the entire tour would be. But now that I've tried it, it's exactly what I've always been looking for ;)

**Reply by FJBDev** [reply]
I'm glad to know it's what you were looking for.
MT is a real "swiss-knife" and I would recommend that you look at and read the [documentation](https://mytourbook.sourceforge.io/mytourbook/index.php/documentation) to discover all the available features

**Reply by ovahead** [reply]
Yeah I know RTFM ;)
Unfortunately, the problem is that the manual is only in English

**Reply by FJBDev** [reply]
>Unfortunately, the problem is that the manual is only in English

Ah, true. But there are browser plugins that can do translations of pages on the fly if needed.

**Reply by FJBDev** [reply]
>Anyway I sent you the files.

After investigation, it seems to be the same issue I noted yesterday, the pauses in the modified file are "malformed" and encapsulated/duplicated and therefore create this negative issue

Example with the file `2023-06-07-16-04-50-fixed.fit`

The 5th pause starts **Wednesday, June 7, 2023 2:10:21 PM** and ends **Wednesday, June 7, 2023 2:26:16 PM**
The 6th pause starts :  **Wednesday, June 7, 2023 2:10:22 PM** and ends **Wednesday, June 7, 2023 2:26:16 PM**

**Comment by FJBDev** [comment]
By the way, I forgot to mention that before but you can easily see the "malformed" pauses when looking at your edited files in the "Tour Pauses" view in MT
See the example below for your file `2023-06-07-16-04-50-fixed.fit`: 

![image](https://github.com/mytourbook/mytourbook/assets/25235665/ede29d92-0e9b-44f5-8d5d-84f5a21ee531)

**Reply by FJBDev** [reply]
Which makes me think that [this feature](https://github.com/mytourbook/mytourbook/issues/1110) could be useful to fix your files with malformed pauses

**Reply by ovahead** [reply]
Most certainly! ;) 

But I'm surprised how these dupe pauses could arise although I haven't changed anything with FFRT. Looks like FFRT add the second (manual) pause automatically, because in the single files is only one manual pause and that is at the beginning of the first file...

**Reply by FJBDev** [reply]
It's very likely that FFRT adds those pauses though I don't have access to their source code nor do i use their tool so it would be best to ask them

**Reply by ovahead** [reply]
Of course, I'll ask him.

**Reply by FJBDev** [reply]
@ovahead 
I have implemented the [pause deletion](https://github.com/mytourbook/mytourbook/pull/1157) and hopefully it will be available in the next version of MT

**Reply by ovahead** [reply]
Very Nice!!
I think the developer of FFRT will now also do something about duplicate pause entries in his tool.
